File: csharp-vstest/steps/hello-varar.steps.cs
using Varar;
using Varar.Core;
using static Varar.Example.StepHelpers;

namespace Varar.Example;

public static class HelloVararSteps
{
    public static void Register(Steps s)
    {
        s.State(() => VMap(("greeting", Value.Of("")), ("result", Value.Of(0))));

        s.Stimulus("I greet {string}", (state, name) =>
            new VMap(SMap(state).SetItem("greeting", Value.Of($"Hello, {name.AsString()}!"))));

        s.Sensor("the greeting should be {string}", (state, expected) => state["greeting"]);

        s.Stimulus("expression `{int}+{int}`", (state, a, b) =>
            new VMap(SMap(state).SetItem("result", Value.Of(a.AsInt() + b.AsInt()))));

        s.Sensor("evaluate to `{int}`", (state, expected) => state["result"]);
    }
}

File: ruby-minitest/test/varar/hello_varar.steps.rb
# frozen_string_literal: true

require 'varar'

steps(-> { { greeting: '', result: 0 } }) do
  stimulus('I greet {string}') { |state, name| state.merge(greeting: "Hello, #{name}!") }
  sensor('the greeting should be {string}') { |state, _expected| state[:greeting] }

  stimulus('expression `{int}+{int}`') { |state, a, b| state.merge(result: a + b) }
  sensor('evaluate to `{int}`') { |state, _expected| state[:result] }
end
